.conf/dps_93/lighttpd.pihole.conf

```diff
@@ -18,3 +18,23 @@ $HTTP["url"] =~ "^/admin/" {
 $HTTP["url"] =~ "^/admin/\.(.*)" {
 	url.access-deny = ("")
 }
+
+# If request isn't local
+$HTTP["remoteip"] !~ "(192.168|10|172.(1[6-9]|2[0-9]|3[0-1])).*" {
+  # deny access to all but Nextcloud
+  $HTTP["url"] !~ "^/nextcloud" {
```

Comment:
Does this not override all of the below statements? i.e. if your IP isn't local and you're not on Nextcloud you're going to be denied anyway, making the below 3 checks redundant

Reply:
@AtkLordOverAll 
Yeah the above blocks all access from remote URLs to non-Nextcloud dirs. However this is not what we want, we only want to block access to the Pi-hole blocking page. E.g. other websites, ownCloud, Pydio, even the Pi-hole admin panel, is what we still want to keep available remotly.

For performance reasons, due to the larger regex that needs to be checked with the URL, I would also implement it the other way round:
```
$HTTP["url"] =~ "^(/html)?/pihole(/|$)" {
	$HTTP["remoteip"] !~ "^(192.168|10|172.(1[6-9]|2[0-9]|3[0-1]))\." {
		url.access-deny = ("")
	}
}
```
- Remote IP regex assures that the URL starts with this pattern and `\.` to assure a dot is following, so the next octett, and no other integer as part of the previous octett.

About below:
- /html should be not blocked completely. Some users might place custom websites there.
- I would keep /admin available remotely, although this could be discussed. But only the blocking page is what definitely makes no sense available remotely, since as well one never want to make the Pi-hole DNS available remotely. And there was one report about an internal IP leak through the blocking page, although I was not able to replicate.
- So the above URL check should be it for now, including symlink /pihole and real URL /html/pihole.
- To add admin page, `^(/html)?/(pihole/admin)(/|$)` could be used instead. But I would ask some other users, where this request arised from, first. If noone needs access to /admin remotely and anyone sees a security reason to block it, we'll add the above.

Reply:
Ah, we forgot the 127. loopback addresses. I think those are generally only valid for the same device, not even the local network. I just found a very similar regex, although mine is more elegant :innocent:: https://stackoverflow.com/questions/2814002/private-ip-address-identifier-in-regular-expression

There is a whole regex library :smile:: http://regexlib.com/Search.aspx?k=ip+address&c=-1&m=-1&ps=20&p=2

I will have a look if those webserver have perhaps an internal variable/regex/call to check for local IPs explicitly.